indexer-app/src/cmr/indexer/services/index_service.clj

```diff
@@ -174,6 +176,121 @@
+                              (map #(dissoc % :priority) h))
+              suggestion-docs (->> value-map
+                                   (map #(suggestion-doc concept-id key-name %))
+                                   (remove nil?))]]
+    suggestion-docs))
+
+(defn- positions
+  "Return a map of the index of each item that matches the predicate"
+  [pred coll]
+  (keep-indexed (fn [idx x]
+                  (when (pred x)
+                    idx))
+                coll))
+
+(defn- science-keywords->elastic-docs
+  "Convert hierarchical science-keywords to colon-separated elastic docs for indexing"
+  [science-keywords keyword-hierarchy]
+  (for [sk-map science-keywords
+         :let [values (->> keyword-hierarchy
+                           (map #(get sk-map %))
+                           (remove nil?))
+               ; take the leaf keyword and store it as the field name
+               terminal-key (->> keyword-hierarchy
+                                 (map #(get sk-map %))
+                                 (positions nil?))
+               ; no nil items in `terminal-key` simply means that there were no
+               ; keys in `keyword-hierarchy` that were not present in the
+               ; science-keywords object
+               keyword-field (if (empty? terminal-key)
+                               (last keyword-hierarchy)
+                               (->> terminal-key
+                                    first
+                                    (nth keyword-hierarchy)
+                                    name))]]
+     {:type "science-keywords"
+       :value (s/join ":" values)
+       :field keyword-field
+       :_index "1_autocomplete"
+       :_type "suggestion"}))
+
+(defn- get-science-keyword-elastic-docs
+  "Fetch science keywords and marshall them into indexable format"
+  [context]
+  (let [kms-keywords (kms/get-keywords-for-keyword-scheme context :science-keywords)
+        science-keywords (map #(dissoc % :uuid) kms-keywords)
+        keyword-hierarchy [:topic :category :variable-level-1 :variable-level-2
+                           :variable-level-3 :detailed-variable]]
+    (science-keywords->elastic-docs science-keywords keyword-hierarchy)))
+
+(defn- reindex-science-keyword-suggestions
+  "Reindex all science keywords for the suggestion index"
+  [context]
+  (let [science-keyword-docs (get-science-keyword-elastic-docs context)]
+    (es/bulk-index-autocomplete-suggestions context science-keyword-docs)))
+
+(defn- collection->suggestion-doc
+  "Convert collection concept metadata to UMM-C and pull facet fields
+  to be indexed as autocomplete suggestion doc"
+  [context collection]
+  (let [concept-id (:concept-id collection)
+        parsed-concept (cp/parse-concept context collection)
+        humanized-fields (humanizer/collection-humanizers-elastic context parsed-concept)
+        suggestion-docs (get-suggestion-docs humanized-fields concept-id)]
```

Comment:
Since get-suggestion-docs doesn't need concept-id, you can get rid of it on line 262 and don't pass it on line 265